### DISCUSSION TITLE
RulesEngine constructors, breaking change

### DISCUSSION THREAD

**Discussion by ivansager** [microsoft/RulesEngine#400] [Q&A] [answered]
The latest release has a breaking change where ILogger has been removed from RulesEngine and all its constructors. The documentation still shows the requirement of the 3rd parameter.

Before we refactor our implementation, I wanted to be sure that the removal of ILogger from the constructor is intentional and is the correct constructor signature going forward.

**Comment by abbasc52** [comment, marked as answer]
Hi @ivansager, the removal of ILogger is intentional and is documented in https://github.com/microsoft/RulesEngine/blob/main/CHANGELOG.md

It looks like there are some places in documentation which still needs updating. I will have a look at the documentation where it might be wrong

**Reply by abbasc52** [reply]
As for the reason of removal, ILogger was not used much internally, and we wanted to be consistent on how we return error/information.
